### PR TITLE
Modal positioning

### PR DESCRIPTION
**Pull request by Kajish**

### CODE REVIEW

File: src/components/charts/BarChart.tsx [hunk 1]
```diff
@@ -330,7 +331,7 @@ const BarChart = () => {
                           title={form.title}
                           containerclassname="calendarSessions"
                           titleclassname="top-5"
-                          imageclassname="h-[20px] w-[20px] relative top-[1.75rem] left-[0.5vw] z-[1]"
+                          imageclassname="h-[20px] w-[20px] relative top-[1.75rem] left-[0.75vw] z-[1]"
```
Comment (sarath-sasi-panickaveetil) [inline src/components/charts/BarChart.tsx]: > h-[20px] can be replaced by h-5
> same comment for width too

File: src/components/charts/BarChart.tsx [hunk 2]
```diff
@@ -495,7 +493,7 @@ const BarChart = () => {
                       showTime
                       containerclassname="calendarSessions"
                       titleclassname="top-5"
-                      imageclassname="h-[20px] w-[20px] relative top-[1.75rem] md:top-[3vh] left-[0.5vw] z-[1]"
+                      imageclassname="h-[20px] w-[20px] relative top-[2rem] md:top-[3vh] left-[2vw] z-[1]"
```
Comment (sarath-sasi-panickaveetil) [inline src/components/charts/BarChart.tsx]: same comment as above. For Height, width and top, use default tailwind classes

File: src/index.css [hunk 3]
```diff
@@ -61,6 +61,24 @@
   @apply h-[13px] rotate-180;
 }
 
+.home-sessions {
+  @apply pt-[3rem] pb-[4rem] sm:py-8;
```
Comment (sarath-sasi-panickaveetil) [inline src/index.css]: use pt-12 and pb-16

File: src/index.css [hunk 4]
```diff
@@ -165,14 +184,17 @@ button.p-button-secondary {
...
+}
 .filtersModal .p-dialog-header .p-dialog-title {
   @apply font-normal text-center text-[16px];
 }
 .filtersModal .p-dialog-content {
   @apply pt-[20px];
 }
 .opmFilters .p-dropdown-label {
-  @apply pt-[7px];
+  @apply pt-[8px];
```
Comment (sarath-sasi-panickaveetil) [inline src/index.css]: pt-2

File: src/index.css [hunk 5]
```diff
@@ -418,7 +426,7 @@ button.p-button-secondary {
   @apply py-4 text-gray-101 z-10 font-medium bg-black-200 border-black-400 border-solid border-b;
 }
 .custom-popup .p-dialog-content {
-  @apply pt-4 overflow-auto text-white-600 bg-black-200 pl-[24px] mr-[10px] pr-[14px];
+  @apply sm:pt-6 overflow-auto text-white-600 bg-black-200 pl-[24px] mr-[10px] pr-[14px];
```
Comment (sarath-sasi-panickaveetil) [inline src/index.css]: use tailwind default classes. pl-6, mr-2.5, pr-3.5

File: src/views/opm.tsx [hunk 6]
```diff
@@ -472,31 +472,31 @@ const OPM: React.FC = () => {
...
                           onChange={(event) => handleFormChange(event)}
-                          className="border rounded-[8px] border-solid border-slate-300 border-1 h-[38px] sm:w-[20vw] lg:w-[11vw]"
                         />
                       )}
                       {form.type === INPUT_TYPES.time && (
                         <CustomCalendar
                           name={form.name}
-                          containerclassname="calendarOpmComparison ml-[10px] lg:w-[10vw] lg:w-[12vw] xl:w-[14vw] sm:mr-[-0.25rem]"
-                          titleclassname="top-5"
+                          titleclassname="top-[1.25rem]"
```
Comment (sarath-sasi-panickaveetil) [inline src/views/opm.tsx]: top-5

File: src/views/opm.tsx [hunk 7]
```diff
@@ -540,77 +540,86 @@ const OPM: React.FC = () => {
...
+                                  onChange={(event) => handleFormChange(event)}
+                                  className="h-[39px]"
+                                />
+                              )}
+                              {form.type === INPUT_TYPES.time && (
+                               
+                                  <CustomCalendar
+                                    name={form.name}
+                                     containerclassname="opmFiltersMobileCalendar"
+                                    imageclassname="h-[20px] w-[20px] top-[1.75rem] md:top-[3vh] left-[2vw] z-[1]"
```
Comment (sarath-sasi-panickaveetil) [inline src/views/opm.tsx]: use default classes

File: src/views/opmComparison.tsx [hunk 8]
```diff
@@ -495,65 +495,56 @@ const OpmComparison: React.FC = () => {
...
-                              form.name === "endDate"
-                                ? CURRENT_PST_DATE
-                                : null
-                            }
-                          />
+                           <CustomCalendar
+                           name={form.name}
+                           titleclassname="top-[1.25rem]"
+                           containerclassname="lg:max-w-[11rem]"
+                           imageclassname="h-[20px] w-[20px] relative top-[1.75rem] left-[0.75vw] z-[1]"
```
Comment (sarath-sasi-panickaveetil) [inline src/views/opmComparison.tsx]: use default classes

File: src/views/opmComparison.tsx [hunk 9]
```diff
@@ -599,8 +587,8 @@ const OpmComparison: React.FC = () => {
                               <CustomCalendar
                                 name={form.name}
                                 containerclassname="opmFiltersMobileCalendar"
-                                imageclassname="h-[20px] w-[20px] relative top-[1.75rem] md:top-[3vh] left-[3.5vw] z-[1]"
-                                titleclassname="left-[2vw] md:left-[0] top-5"
+                                imageclassname="h-[20px] w-[20px] relative top-[1.75rem] md:top-[3vh] left-[2vw] z-[1]"
```
Comment (sarath-sasi-panickaveetil) [inline src/views/opmComparison.tsx]: use default classes

File: src/components/appcontent.tsx [hunk 10]
```diff
@@ -22,7 +22,7 @@ const AppContent: FC<Props> = ({ showSidePane, showNavbar, appContent }) => {
   const [selectedMenu, setSelectedMenu] = useState<number>(0);
   const location = useLocation();
   const IS_FULLSCREEN = location?.pathname.includes("fullscreen");
-  const IS_ORDER_DETAILS = location?.pathname.includes("orderDetails");
+  const IS_ORDER_DETAILS = location?.pathname.includes("order-details");
```
Comment (raj-kumar-litmus) [inline src/components/appcontent.tsx]: move this to constants file.

File: src/components/common/CustomCalendar.tsx [hunk 11]
```diff
@@ -94,16 +94,18 @@ const CustomCalendar: FC<CustomCalendarProps> = (props) => {
...
+    <div className={`relative ${props.containerclassname} w-full self-end`}>
       <div
-        className={`text-xs font-medium pb-1 text-gray-400 relative ${props.titleclassname}`}
+        className={`text-xs font-medium pb-1 text-gray-400 relative w-full ${props.titleclassname}`}
       >
         {props.title}
       </div>
-      <img className={props.imageclassname} src={props.imgsrc} />
-      <Calendar
+      <img className={`absolute ${props.imageclassname}`} src={props.imgsrc} />
```
Comment (raj-kumar-litmus) [inline src/components/common/CustomCalendar.tsx:103]: Use image component.

File: src/index.css [hunk 12]
```diff
@@ -165,14 +184,17 @@ button.p-button-secondary {
 .filtersModal .p-dialog-header {
   @apply px-[8px] py-[10px] rounded-t-[10px] border-b border-b-gray-700 border-solid;
 }
+.filtersModal-popup.p-dialog {
+  @apply absolute max-[639px]:max-h-[calc(100vh-154px)] max-[480px]:w-full w-[390px] max-[480px]:!bottom-0 sm:max-h-[528px];
```
Comment (raj-kumar-litmus) [inline src/index.css]: you have added `480` as a new breakpoint. Use that here.

File: src/components/common/searchbar.tsx [hunk 13]
```diff
@@ -1,28 +1,23 @@
...
-          <CustomImage
-            src={HeaderSearch}
-            alt="Search"
-            className="absolute left-5 bottom-3"
-          />
-        )}
         <CustomInputText
           type="text"
-          className={`w-[15.6rem] h-10 !rounded-full focus:outline-none  bg-gray-900 text-gray-300 placeholder:text-gray-300 placeholder:font-helvetica placeholder:font-normal !shadow-none !border-none ${props.className}`}
+          className={`w-[15.6rem] h-[2.5rem] !rounded-full focus:outline-none  bg-gray-900 text-gray-300 placeholder:text-gray-300 placeholder:font-helvetica placeholder:font-normal !shadow-none !border-none ${props.className}`}
```
Comment (sarath-sasi-panickaveetil) [inline src/components/common/searchbar.tsx]: use h-10

File: src/index.css [hunk 14]
```diff
@@ -61,6 +61,24 @@
   @apply h-13 rotate-180;
 }
 
+.home-sessions {
+  @apply pt-[3rem] pb-[4rem] sm:py-8;
```
Comment (sarath-sasi-panickaveetil) [inline src/index.css]: this is not changed. use pt-12 and pb-16

File: src/index.css [hunk 15]
```diff
@@ -209,7 +231,7 @@ button.p-button-secondary {
   @apply relative left-2.5;
 }
 .opmFiltersMobileCalendar span.p-calendar {
-  @apply relative left-2 h-[39px];
+  @apply relative  h-[2.5rem];
```
Comment (sarath-sasi-panickaveetil) [inline src/index.css]: h-10

File: src/views/opm.tsx [hunk 16]
```diff
@@ -479,149 +479,153 @@ const OPM: React.FC = () => {
...
+                          containerclassname="lg:min-w-11r"
+                          imageclassname="h-5 w-5 relative top-1.75r left-0.5w z-1"
+                          placeholder={DATE_AND_TIME_FORMATS.MM_DD_YYYY_HH_MM}
+                          title={form.label}
+                          showTime={form.showTime}
+                          iconPos={form.iconPos || "left"}
+                          imgsrc={form.imgsrc}
+                          onChange={(event) => handleFormChange(event)}
+                          value={form.value}
+                          maxDate={form.name === "date" ? new Date() : null}
```
Comment (sarath-sasi-panickaveetil) [inline src/views/opm.tsx]: I think this should be current PST instead of new Date(). Please check with Thasnim

File: src/views/opm.tsx [hunk 17]
```diff
@@ -479,149 +479,153 @@ const OPM: React.FC = () => {
...
+                                    iconPos={form.iconPos || "left"}
+                                    imgsrc={form.imgsrc}
+                                    onChange={(event) =>
+                                      handleFormChange(event)
+                                    }
+                                    value={form.value}
+                                    maxDate={
+                                      form.name === "startDate" ||
+                                      form.name === "endDate"
+                                        ? new Date()
```
Comment (sarath-sasi-panickaveetil) [inline src/views/opm.tsx]: Please check current PST date has to be passed here

File: src/components/DropDown.tsx [hunk 18]
```diff
@@ -5,27 +5,30 @@ import { CustomDropdownProps } from "../@types/components/commonTypes";
...
+          className="w-full label-position"         
         />
-      </span>
+      </div>
     </div>
   ) : (
-    <div className="flex flex-col self-end inputContainer md:w-8w">
-      <label className="labelClass relative ml-18">{props.label}</label>
+    <div className="flex flex-col self-end md:w-[8vw]">
+      <label className="labelClass relative ml-[18px]">{props.label}</label>
```
Comment (thasnim-cm) [inline src/components/DropDown.tsx]: 18px can be used from  tailwind config

File: src/components/DropDown.tsx [hunk 19]
```diff
@@ -5,27 +5,30 @@ import { CustomDropdownProps } from "../@types/components/commonTypes";
...
           placeholder={props.placeholder}
+          className="w-full label-position"         
         />
-      </span>
+      </div>
     </div>
   ) : (
-    <div className="flex flex-col self-end inputContainer md:w-8w">
-      <label className="labelClass relative ml-18">{props.label}</label>
+    <div className="flex flex-col self-end md:w-[8vw]">
```
Comment (thasnim-cm) [inline src/components/DropDown.tsx]: use md:w-8w

File: src/components/charts/BarChart.tsx [hunk 20]
```diff
@@ -354,16 +355,16 @@ const BarChart = () => {
                       {form.type === "dropdown" && (
                         <CustomDropdown
                           name={form.name}
-                          title={form.title}
+                          label={form.title}
+                          containerclassname="!min-w-[9rem]"
                           value={form.value}
+                          imageclassname="z-[1]"
```
Comment (thasnim-cm) [inline src/components/charts/BarChart.tsx]: z-1

File: src/index.css [hunk 21]
```diff
@@ -185,7 +207,7 @@ button.p-button-secondary {
 }
 .opmFiltersMobile.p-dialog .p-dialog-header,
 .opmFiltersMobile.p-dialog .p-dialog-content {
-  @apply bg-black-200 border border-solid border-gray-900 pl-[3vw];
+  @apply bg-black-200 border border-solid border-[#30343B] pl-3 pr-3;
```
Comment (thasnim-cm) [inline src/index.css]: replace hexcode with corresponding config value

File: src/views/opm.tsx (hunk 16 above)
Comment (thasnim-cm) [reply src/views/opm.tsx]: maxDate={ form.name === "date" ? CURRENT_PST_DATE : null}

File: src/views/opm.tsx (hunk 17 above)
Comment (thasnim-cm) [reply src/views/opm.tsx]: it should be CURRENT_PST_DATE

**Review by sarath-sasi-panickaveetil — APPROVED** [review]
Looks good to me

**Review by thasnim-cm — APPROVED** [review]

**Pull request merged**